scripts/models/token/token.js

```diff
@@ -45,8 +45,90 @@ class Token {
     return JSON.stringify(serial, null, CONSTANTS.SERIAL_SPECIFICATIONS.INDENT_TAB_SIZE).replaceAll(sortObject.CONSTANTS.SORT_LOCK_CHARACTER, "");
   }
 
+  /**
+   * Validate if the provided Token can be merged into this Token.
+   *
+   * @param {DTO} dto
+   * @param {Token} dto.token Token to validate if it can be merged into this Token.
+   * @returns {{can: Boolean, reason: String}} If the Token can be merged into the provided Token.
+   */
+  canMerge({ token }) {
+    let can = true;
+    let reason;
+
+    if (this.category !== token.category) {
+      can = false;
+      reason = 'token "category" properties do not match';
+    } else if (this.format !== Token.CONSTANTS.TOKEN_FORMATS.STANDARD) {
+      can = false;
+      reason = `token format "${this.format}" for this token cannot be merged, normalize this token first`;
+    } else if (token.format !== Token.CONSTANTS.TOKEN_FORMATS.STANDARD) {
+      can = false;
+      reason = `token format "${token.format}" for this token cannot be merged, normalize the provided token first`;
+    }
+
+    return { can, reason };
+  }
+
+  /**
+   * Merge the provided Token into this Token.
+   *
+   * @param {DTO} dto
+   * @param {Token} dto.token Token to merge into this Token.
+   * @returns {This} This Token merged with new values from the provided Token.
+   */
+  merge({ token }) {
+    const { can, reason } = this.canMerge({ token });
+
+    if (!can) {
+      throw new Error(`models.${this.constructor.name}.merge() :: can not merge the provided token into this token: ${reason}`);
+    }
+
+    this.mergeData({ data: token.data });
+
+    return this;
+  }
+
+  /**
+   * Normalize this Token.
+   *
+   * @returns {This} This Token with its data and format value normalized.
+   */
+  normalize() {
+    this.normalizeData();
+    this.format = Token.CONSTANTS.TOKEN_FORMATS.STANDARD;
+
+    return this;
+  }
+
+  /**
+   * Merge the provided Token data into this Token.
+   *
+   * @protected
+   * @abstract
+   * @param {DTO} dto
+   * @param {Token.Data} dto.data Token data to merge into this token.
+   * @returns {This} This Token merged with the provided data merged into it.
+   */
+  mergeData({ data }) {
+    throw new Error(`models.Token.mergeData() :: abstract method "mergeData()" for class "${this.constructor.name}" has not bee implemented`);
```

Comment:
Good, so this one is pending

Reply:
Correct, a run-time abstraction check is placed here for clear directives when they aren't abstracted correctly.